### PR TITLE
tests: for kv0 overload test, make jemalloc release memory to OS more…

### PR DESCRIPTION
**Pull request by sumeerbhola**
… aggressively

kv0/enc=false/nodes=1/size=64kb/conc=4096 is sometimes OOMing on arm64 on AWS. It is possible that this is due to higher memory usage in the cgo allocations. This change reduces the difference betweem jemalloc resident and allocated bytes by releasing more aggressively to the OS.

Testing this change reduced "9.2 GiB/11 GiB CGO alloc/total" to "9.2 GiB/9.8 GiB CGO alloc/total".

Informs #125769

Epic: none

Release note: None

### CODE REVIEW

File: pkg/cmd/roachtest/tests/kv.go [hunk 1]
```diff
@@ -190,7 +197,15 @@ func registerKV(r registry.Registry) {
...
-		{nodes: 1, cpus: 8, readPercent: 0, concMultiplier: 4096, blockSize: 1 << 16 /* 64 KB */},
+		//
+		// jemallocReleaseFaster is set to true due to OOMs observed in
+		// https://github.com/cockroachdb/cockroach/issues/125769. There is
+		// unavoidable 20% internal fragmentation in the Pebble block cache, since
+		// a 64KB value size causes a 64+KB sstable block, which allocates
+		// from the 80KB size class in jemalloc. So the allocated bytes from jemalloc
+		// by the block cache are 20% higher than configured. By setting this flag to true,
+		// we reduce the (resident-allocated) size in jemalloc.
+		{nodes: 1, cpus: 8, readPercent: 0, concMultiplier: 4096, blockSize: 1 << 16 /* 64 KB */, jemallocReleaseFaster: true},
```
Comment (srosenberg) [inline pkg/cmd/roachtest/tests/kv.go:208]: Do we expect the throughput to drop? Either way, a roachperf annotation would probably be insightful.

**Review by srosenberg — APPROVED** [review]

**Comment by renatolabs** [comment]
Would be nice to test this change by running this particular kv0 benchmark 10 or 20 times on this branch -- it was failing at least once (generally twice) every time when run with a  `--count 10` flag.

Meta question: if it turns out that the way for this test to pass is to tweak malloc, do we want to continue testing this particular configuration?

I'm also curious if we ever found out what started causing these failures. The Pebble bump mentioned in the original issue does seem to have been the point in which they started to happen.

**Review by sumeerbhola — COMMENTED** [review]
> Would be nice to test this change by running this particular kv0 benchmark 10 or 20 times on this branch -- it was failing at least once (generally twice) every time when run with a `--count 10` flag.

Done. Zero failures in 10 runs with this change, and 7 failures in 20 runs without.

> Meta question: if it turns out that the way for this test to pass is to tweak malloc, do we want to continue testing this particular configuration?

Yes, this overload configuration is a test for admission control.  Node liveness used to suffer pre-AC, so it tests something.

> I'm also curious if we ever found out what started causing these failures. The Pebble bump mentioned in the original issue does seem to have been the point in which they started to happen.

The Pebble-bump is not quite explainable (see the third bullet in https://github.com/cockroachdb/cockroach/issues/125769#issuecomment-2206451050), but there are so many layers involved that there is guesswork.

TFTRs!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/126930#-:-O1Wwg3d5QqAk93X0DrE:bafpa01)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @herkolategan, @renatolabs, and @srosenberg)

___
*[`pkg/cmd/roachtest/tests/kv.go` line 104 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/126930#gh-1672423196:-O1TDG22EDgV6nK1h67z:b6oldby) ([raw file](https://github.com/cockroachdb/cockroach/blob/5bd38d67e3d507843483cdf40d5f3c030604124b/pkg/cmd/roachtest/tests/kv.go#L104)):*
<details><summary><i>Previously, srosenberg (Stan Rosenberg) wrote…</i></summary><blockquote>

Should we maybe cap the number of background threads? By default it's number of cpus.
</blockquote></details>

I have no experience with tuning the number of such threads, so I'd prefer to just start with this simple thing.
___
*[`pkg/cmd/roachtest/tests/kv.go` line 208 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/126930#gh-1672425112:-O1TEwzf3cYX9gXmev_V:b-qjnclh) ([raw file](https://github.com/cockroachdb/cockroach/blob/5bd38d67e3d507843483cdf40d5f3c030604124b/pkg/cmd/roachtest/tests/kv.go#L208)):*
<details><summary><i>Previously, srosenberg (Stan Rosenberg) wrote…</i></summary><blockquote>

Do we expect the throughput to drop? Either way, a roachperf annotation would probably be insightful.
</blockquote></details>

Good point. I forgot to check that, though this test is not CPU bound, so would be surprising. I can merge this and add an annotation if I see throughput drop over the next couple of days.

**Comment by sumeerbhola** [comment]
bors r=srosenberg,renatolabs

**Comment by sumeerbhola** [comment]
> I can merge this and add an annotation if I see throughput drop over the next couple of days.

I checked roachperf and there was no throughput drop.

**Comment by renatolabs** [comment]
blathers backport 24.2

**Pull request merged**